### ISSUE TITLE
BulkInsert not working

### ISSUE THREAD

**Issue by Woudjee** [borisdj/EFCore.BulkExtensions#1349]
Why does this work?

`List<BrepEntity> brepjes = new List<BrepEntity>();

for (int i = 0; i < 100; i++)
{
    brepjes.Add(new BrepEntity()
    {
        Polygons = new List<PolygonEntity>()
        {
            new PolygonEntity()
            {
                Vertices = new List<VectorEntity>()
                {
                    new VectorEntity() { X = 123, Y = 2348.324f, Z = 234293784.1f },
                }
            },
            new PolygonEntity()
            {
                Vertices = new List<VectorEntity>()
                {
                    new VectorEntity() { X = 334, Y = 244648.324f, Z = 909793784.1f },
                }
            }
        }
    });
}

Stopwatch stopwatch = Stopwatch.StartNew();
_context.BulkInsert(brepjes, options => options.IncludeGraph = true);
_context.BulkSaveChanges();

stopwatch.Stop();
return stopwatch.ElapsedMilliseconds;`

But this does not?

`List<GridInfoEntity> test = new List<GridInfoEntity>()
{
    new GridInfoEntity()
    {
        GridInfoX = new List<GridInfoXEntity>()
        {
            new GridInfoXEntity("Label", 123),
        },
        GridInfoY = new List<GridInfoYEntity>()
        {
            new GridInfoYEntity("Label", 456),
        },
        GridInfoZ = new List<GridInfoZEntity>()
        {
            new GridInfoZEntity("Label", 789),
        },
    }
};

Stopwatch stopwatch = Stopwatch.StartNew();
_context.BulkInsert(test, options => options.IncludeGraph = true);
_context.BulkSaveChanges();

stopwatch.Stop();
return stopwatch.ElapsedMilliseconds;`

I am getting this error, which is incorrect:

Missing Column : GridInfoId
On entity : GridInfoXEntity
On Table : [Domain].[GridInfo]

**Comment by borisdj** [comment]
Could you add entire code of class _GridInfoEntity_ and _GridInfoZEntity_ or write a Test for the issue.

**Comment by Woudjee** [comment]
Yes, of couse. See below (screenshot for readability + copy paste):

**GridInfoEntity**
![image](https://github.com/borisdj/EFCore.BulkExtensions/assets/84737524/8b4939c2-598a-4e95-8875-b100c63230a5)

[Table("GridInfo", Schema = "Domain")]
public class GridInfoEntity : Entity
{
    public virtual ICollection<GridInfoXEntity> GridInfoX { get; set; }
    public virtual ICollection<GridInfoYEntity> GridInfoY { get; set; }
    public virtual ICollection<GridInfoZEntity> GridInfoZ { get; set; }
}

**GridInfoXEntity**
![image](https://github.com/borisdj/EFCore.BulkExtensions/assets/84737524/b86f2383-4528-4432-bbbb-4e7d61efb880)

[Table("GridInfo.X", Schema = "Domain")]
public class GridInfoXEntity : Entity
{
    public GridInfoXEntity(string label, float value)
    {
        ValueX = value;
        Label = label;
    }

    public GridInfoXEntity()
    {

    }

    public float ValueX { get; set; }
    public string Label { get; set; }

    public virtual GridInfoEntity GridInfo { get; set; }
}

Exactly the same applies to the Y and Z variant (I double checked in my code).

And for the Brep-chain:

**BrepEntity**
![image](https://github.com/borisdj/EFCore.BulkExtensions/assets/84737524/15c9be5b-b859-403c-8e95-5c09b7b7a57a)

public class BrepEntity : Entity
{
    public virtual ICollection<PolygonEntity> Polygons { get; set; }
}

**PolygonEntity**
![image](https://github.com/borisdj/EFCore.BulkExtensions/assets/84737524/57cc1caa-4b48-4ef7-bc16-06ad1ce5bc6e)

 public class PolygonEntity : Entity
 {
     public int BrepId { get; set; }
     public virtual BrepEntity Brep { get; set; }
     public virtual ICollection<VectorEntity> Vertices { get; set; }
 }

**VectorEntity**
![image](https://github.com/borisdj/EFCore.BulkExtensions/assets/84737524/9ed45171-3215-4208-b43d-a3b8b86803e3)

public class VectorEntity : Entity
{
    public float X { get; set; }
    public float Y { get; set; }
    public float Z { get; set; }

    public int PolygonId { get; set; }
    public virtual PolygonEntity Polygon { get; set; }
}

**Comment by borisdj** [comment]
Schemas are actually supported with main Bulk ops.
But _IncludeGraph_ often had different exceptions.
I lookup into it more.
In the meantime you could try to call 2 times BulkInsert, instead of using IncludeGraph.

**Comment by borisdj** [comment]
Yes, that's right.

**Comment by Woudjee** [comment]
Thank you Boris for you quick replies. I will keep an eye on this post and would be more than happy to contribute to solving this problem wherever I can. Just let me know.
